### ISSUE TITLE
Userid event vars treated as entity indexes

### ISSUE THREAD

**Issue by jotalanusse** [saul/demofile-net#12]
This error pops up when trying to load a CS2 match from a `.dem` file:
```
System.IndexOutOfRangeException: 'Index was outside the bounds of the array.'
```

Example code used:
```csharp
﻿using DemoFile;

internal class Program
{
    public static async Task Main(string[] args)
    {
        var path = args.SingleOrDefault() ?? throw new Exception("Expected a single argument: <path to .dem>");

        var demo = new DemoParser();
        demo.Source1GameEvents.PlayerDeath += e =>
        {
            Console.WriteLine($"{e.Attacker?.PlayerName} [{e.Weapon}] {e.Player?.PlayerName}");
        };

        await demo.Start(File.OpenRead(path));

        Console.WriteLine("\nFinished!");
    }
}
```

The error seems to appear when trying to access the `e.Player` or `e.Attacker` properties.

I'm in WinForms using version NET version `.NET 7.0`.

I'm leaving a link to the file used [here](https://files.catbox.moe/zgfr0v.dem).

**Comment by saul** [comment]
Thanks for reporting this.

The issue was that the userids were incorrectly being interpreted as entity indexes. I've just pushed a fix that is available as v0.2.7

I've tested the new version of the library with the demo that you linked. Let me know if the issue is unresolved.

**Comment by jotalanusse** [comment]
Awesome, seems to be working now, thanks!

Keep up the good work, very nice library to have in C#.
